### PR TITLE
Be more defensive in redirect code

### PR DESCRIPTION
**Pull request by ericholscher**
Fixes https://github.com/readthedocs/readthedocs.org/issues/11338

### CODE REVIEW

**Review by humitos — APPROVED** [review]
Looks good to me. It would be good if we can write a test case for this case.

File: readthedocs/rtd_tests/tests/test_footer.py [hunk 1]
```diff
@@ -318,6 +318,12 @@ def test_not_built_versions(self):
         self.assertIn("/en/latest/", response.data["html"])
         self.assertNotIn("/en/2.0/", response.data["html"])
 
+    def test_invalid_footer_url(self):
+        # The built versions appears on the footer
+        self.url = reverse("footer_html") + "/invalid/url"
+        response = self.render()
+        self.assertEqual(response.status_code, 404)
```
Comment (humitos) [inline readthedocs/rtd_tests/tests/test_footer.py:325]: This test doesn't seem to be related with the change you've done. This test passes with _and_ without the code you added in `ServeDocsBase`

Comment (humitos) [reply readthedocs/rtd_tests/tests/test_footer.py:325]:
```
root@a8d91d542ef9:/usr/src/app/checkouts/readthedocs.org# git diff
diff --git a/common b/common
--- a/common
+++ b/common
@@ -1 +1 @@
-Subproject commit d83c1d569e2c4f2873a5542352f8b2db38d18d11
+Subproject commit d83c1d569e2c4f2873a5542352f8b2db38d18d11-dirty
diff --git a/readthedocs/proxito/views/serve.py b/readthedocs/proxito/views/serve.py
index af1ce2ad8..b27280143 100644
--- a/readthedocs/proxito/views/serve.py
+++ b/readthedocs/proxito/views/serve.py
@@ -123,8 +123,8 @@ class ServeDocsBase(CDNCacheControlMixin, ServeRedirectMixin, ServeDocsMixin, Vi
         """
         unresolved_domain = request.unresolved_domain
         # Protect against bad requests to API hosts that don't set this attribute.
-        if not unresolved_domain:
-            raise Http404
+        # if not unresolved_domain:
+        #     raise Http404
         # Handle requests that need canonicalizing first,
         # e.g. HTTP -> HTTPS, redirect to canonical domain, etc.
         # We run this here to reduce work we need to do on easily cached responses.
root@a8d91d542ef9:/usr/src/app/checkouts/readthedocs.org# tox -e py310 -- --nomigrations -k test_invalid_footer_url
py310: commands[0]> sh -c 'export DJANGO_SETTINGS_MODULE=readthedocs.settings.test; pytest  --pyargs readthedocs --suppress-no-test-exit-code -m "not proxito and not embed_api" --nomigrations -k test_invalid_footer_url'
========================================================================================== test session starts ==========================================================================================
platform linux -- Python 3.10.12, pytest-8.2.2, pluggy-1.5.0
cachedir: .tox/py310/.pytest_cache
django: version: 4.2.13, settings: readthedocs.settings.test (from env)
rootdir: /usr/src/app/checkouts/readthedocs.org
configfile: pytest.ini
plugins: requests-mock-1.12.1, django-4.8.0, cov-5.0.0, mock-3.14.0, custom-exit-code-0.3.0
collected 2388 items / 2386 deselected / 2 selected                                                                                                                                                     

readthedocs/proxito/tests/test_proxied_api.py .                                                                                                                                                   [ 50%]
readthedocs/rtd_tests/tests/test_footer.py .                                                                                                                                                      [100%]

================================================================================== 2 passed, 2386 deselected in 3.11s ===================================================================================
py310: commands[1]> sh -c 'export DJANGO_SETTINGS_MODULE=readthedocs.settings.proxito.test; pytest  --pyargs readthedocs -m proxito --suppress-no-test-exit-code --nomigrations -k test_invalid_footer_url'
========================================================================================== test session starts ==========================================================================================
platform linux -- Python 3.10.12, pytest-8.2.2, pluggy-1.5.0
cachedir: .tox/py310/.pytest_cache
django: version: 4.2.13, settings: readthedocs.settings.proxito.test (from env)
rootdir: /usr/src/app/checkouts/readthedocs.org
configfile: pytest.ini
plugins: requests-mock-1.12.1, django-4.8.0, cov-5.0.0, mock-3.14.0, custom-exit-code-0.3.0
collected 2388 items / 2388 deselected / 0 selected                                                                                                                                                     

======================================================================================= 2388 deselected in 2.08s ========================================================================================
  py310: OK (9.86=setup[0.05]+cmd[5.21,4.59] seconds)
  congratulations :) (9.92 seconds)
```

Comment (ericholscher) [reply readthedocs/rtd_tests/tests/test_footer.py:325]: It's definitely related, but I think it's probably happening on the 404 handler on the API views because we're skipping adding this variable in the middleware.

Comment (humitos) [reply readthedocs/rtd_tests/tests/test_footer.py:325]: Well, my point was that the test passes with or without the code change from this PR. So, the test has no effect over the code. If we change or remove the code you added by mistake, the test will not cover us.
